lib/screens/stake_screen/stake_screen.dart

```diff
@@ -0,0 +1,340 @@
+
+              ],
+            ),
+          )
+        ],
+      ),
+    );
+  }
+
+  Padding _buildSteps() {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 75),
+      child: Row(
+        children: [
+          Container(
+            padding: EdgeInsets.all(8),
+            child: Text(
+              '1',
+              style: MyStyles.whiteMediumTextStyle,
+            ),
+            decoration:
+                BoxDecoration(gradient: gradient, shape: BoxShape.circle),
+          ),
+          Expanded(
+              child: Container(
+                  height: 3, decoration: BoxDecoration(gradient: gradient))),
+          Container(
+            padding: EdgeInsets.all(8),
+            child: Text(
+              '2',
+              style: MyStyles.whiteMediumTextStyle,
+            ),
+            decoration: BoxDecoration(
+                color: Color(MyColors.Button_BG_Black), shape: BoxShape.circle),
+          )
+        ],
+      ),
+    );
+  }
+
+  Widget _buildStakeApproveButton() {
+    return AnimatedCrossFade(
+        firstChild: Row(
+          children: [
+            Expanded(
+              child: FilledGradientSelectionButton(
+                onPressed: () {
+                  setState(() {
+                    _stakeState = StakesStates.pendingApprove;
+                    _showToast = true;
+                  });
+                  Future.delayed(Duration(seconds: 3), () {
+                    setState(() {
+                      _stakeState = StakesStates.canStake;
+                    });
+                  });
+                },
+                label: 'APPROVE',
+                textStyle: MyStyles.blackMediumTextStyle,
+                gradient: gradient,
+              ),
+            ),
+            if (_stakeState != StakesStates.canStake)
+              SizedBox(
+                width: 15,
+              ),
+            Expanded(
+              child: SizedBox(
+                child: SelectionButton(
+                  gradient: gradient,
+                  child: _stakeState == StakesStates.pendingApprove
+                      ? SizedBox(
+                          height: 20,
+                          width: 20,
+                          child: CircularProgressIndicator(
+                            valueColor:
+                                AlwaysStoppedAnimation<Color>(Colors.white),
+                            strokeWidth: 2,
+                          ),
+                        )
+                      : null,
+                  onPressed: (bool) {},
+                  label: 'STAKE',
+                  selected: false,
+                  textStyle: _stakeState == StakesStates.canStake
+                      ? MyStyles.blackMediumTextStyle
+                      : MyStyles.lightWhiteMediumTextStyle,
+                ),
+              ),
+            )
+          ],
+        ),
```

Comment:
Ich sehe auf den ersten Blick nicht, wie sich first und secondChild unterscheiden. Davon ist sehr viel identisch.

- [ ] firstChild in eine neue _build... Funktion mit aussagekräftigem Namen
- [ ] secondChild in eine neue _build... Funktion mit aussagekräftigem Namen

Reply:
Nach wie vor offen